banyan/src/forest/read.rs

```diff
@@ -41,6 +41,8 @@ struct TraverseState<T: TreeTypes> {
     // Branches can not have zero children, so when this is empty we know that we have
     // to initialize it.
     filter: SmallVec<[bool; 64]>,
+
+    branch: Option<Branch<T>>,
```

Comment:
A comment would be nice.

Reply:
Added one.

But I am actually on the fence regarding this one. Imagine an iterator that is stuck somehow, e.g. due to backpressure. It will hold the "stem" in memory no matter what the cache limits say. Now, have a lot of such iterators, and you are going to run OOM. I guess lots of cache misses in that scenario is preferable to going OOM.

So I think what is needed is a more fine grained control on the number of children for branches, allowing having fewer children for level 0 branches, so you don't end up with huge level 0 branches in cases where the tags dominate the data...

Reply:
If the cache would work as intended, we wouldn't have this discussion, right? So maybe the cache needs to be better tuned, or we need more diagnostics, ..

Reply:
Well, I think the LRU heuristic is not optimal for certain access patterns. We might need something smarter, but I don't think this is urgent enough to hold back cosmos development.

This branch has some definite improvements and some nice tests to prevent regressions, so I would vote to merge it now, and I will look into this on Friday.

Reply:
Good. Maybe raise a ticket.

Reply:
Not quite the same issue, but while looking into this I will also look into the cache thing. So I think it is sufficient:

https://github.com/Actyx/banyan/issues/87